### ISSUE TITLE
Paragraph: Add CSS columns

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#27118]
It would help readability of paragraphs on wide screens if the text were to go to more than one column. This can be achieve using the css `columns` property. It would look like this:

<img width="854" alt="Screenshot 2020-11-19 at 18 01 05" src="https://user-images.githubusercontent.com/275961/99705247-60f18e80-2a91-11eb-8775-08020ef0a167.png">

I think this could be enabled/disabled with a toggle. On wider screens the number of columns would increase, and on mobile we wouldn't have any.

**Comment by aristath** [comment]
This can already be accomplished in themes using block-styles.
The block-style adds the class, and then the theme can simply style that. I don't think we need separate controls etc on the paragraph block, this is something that will need to be considered on the theme level

**Comment by scruffian** [comment]
This feels like something that lots of themes might want to implement, so I'd rather not move it to the theme as we'll need to write the same implementation in many themes. I also think it's something that users will want to customize which is why I'd rather see it as an option in the block.

**Comment by ntsekouras** [comment]
Related: https://github.com/WordPress/gutenberg/issues/25459

**Comment by kjellr** [comment]
> This can already be accomplished in themes using block-styles.

That's true, but I could see the benefit to having additional controls beyond that: a way to adjust the number of columns, and a checkbox for whether or not the columns should disappear on mobile. Those options aren't really suited well to a block style.

**Comment by aristath** [comment]
Did we not use to have a `text-columns` block? AFAIR it was later abstracted and converted to the more generic `columns` block.
I understand there's a difference between using the CSS `columns` property and actually having 2 columns, but usability-wise I don't think it will make much of a difference. If the users wants 2 columns, they just add 2 columns... There's no need for a columns control in paragraphs when we already have that functionality in the columns block. 
However, if the user wants auto-scaling columns, then things change. I can definitely imagine a newspaper-like or heavy text-based article benefiting from that... But in that case it wouldn't define the number of columns, they can already do that by adding columns and splitting their text in separate columns. They could however define the column-width. So something like this would work beautifully in this case:
```css
p {
  column-gap: 2em;
  column-width: 15em;
  column-rule: 1px solid rgba(0,0,0,.1);
}
```
It's responsive, works for all content and viewports and it's pretty simple to implement.
So then the question becomes do we want to allow customizing the column-width? If yes, then we add a control for the width. If not, then this can be a block-style. And if a block-style, then 15em is a pretty reasonable middle-ground for the columns-width :wink:

**Comment by scruffian** [comment]
Can we apply a block style to all blocks using theme.json?

**Comment by aristath** [comment]
I don't think so... but having a `defaultStyle` prop in `theme.json` would be pretty amazing. Perhaps we should create a new issue for that?

**Comment by t-hamano** [comment]
Now that we have `textColumn`s block support, this is easy to achieve. Do we like to close this issue?

theme.json

```json
{
	"version": 3,
	"settings": {
		"appearanceTools": true,
		"layout": {
			"contentSize": "840px",
			"wideSize": "1100px"
		},
		"typography": {
			"textColumns": true
		}
	}
}
```

functions.php

```php
function add_text_columns_support_to_paragraph( $metadata ) {
	if ( ! isset( $metadata['name'] ) || 'core/paragraph' !== $metadata['name'] ) {
		return $metadata;
	}
	if ( ! isset( $metadata['supports'] ) ) {
		$metadata['supports'] = array();
	}
	if ( ! isset( $metadata['supports']['typography'] ) ) {
		$metadata['supports']['typography'] = array();
	}
	$metadata['supports']['typography']['textColumns'] = true;

	return $metadata;
}
add_filter( 'block_type_metadata', 'add_text_columns_support_to_paragraph' );
```

**Comment by scruffian** [comment]
Should we consider enabling it in all paragraph blocks?

**Comment by t-hamano** [comment]
I think adding this support to the Paragraph block is a good start. It works well even when combined with other block supports.

<img width="1234" height="664" alt="Image" src="https://github.com/user-attachments/assets/1f38bc1d-11dc-4a92-8155-5c9c6d6af641" />

**Comment by scruffian** [comment]
I created a PR for this: [Paragraph: Add text column support](https://github.com/WordPress/gutenberg/pull/74656/changes#top)
